src/services/api/credentials.ts

```diff
@@ -72,6 +78,54 @@ export class Credentials {
 
 		const statusOptions = credentialStatus || null;
 
+		if (statusOptions) {
+			const { success, data: statusRegistry } = await CredentialStatusService.instance.getStatusList(
+				statusOptions,
+				customer,
+				true
+			);
+			if (!success || !statusRegistry) {
+				throw new Error('Status Registry Not Found');
+			}
+
+			if (statusRegistry.state === StatusRegistryState.Full) {
+				throw new Error('Status Registry is Full');
+			}
+
+			if (statusRegistry.state !== StatusRegistryState.Active) {
+				throw new Error(`Status Registry is not Active. Current state: ${statusRegistry.state}`);
+			}
+
+			// assign next index in status list
+			const index = statusRegistry.lastAssignedIndex + 1;
+			statusOptions.statusListIndex = index;
```

Comment:
Does this ensure that the response has correct index in the "credentialStatus"? what if index is passed in "credentialStatus"?